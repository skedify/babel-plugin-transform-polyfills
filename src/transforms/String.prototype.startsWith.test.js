describe('String.prototype.startsWith', () => {
  it('should replace String.prototype.startsWith with a polyfill', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.startsWith('a') === true) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.startsWith('a')) {
        console.log(text.startsWith('a'))
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.substr(5).startsWith('a')) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    return expect(result).toMatchSnapshot()
  })
})
