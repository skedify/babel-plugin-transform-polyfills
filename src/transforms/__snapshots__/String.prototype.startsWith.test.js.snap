// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String.prototype.startsWith should not import polyfill more than once 1`] = `
"import _String_prototype_startsWith from 'babel-plugin-transform-polyfills/polyfills/String/prototype/startsWith';

const text = 'this is a string';
if (_String_prototype_startsWith(text, 'a')) {
  console.log(_String_prototype_startsWith(text, 'a'));
} else {
  console.log('failure');
}"
`;

exports[`String.prototype.startsWith should replace String.prototype.startsWith with a polyfill 1`] = `
"import _String_prototype_startsWith from 'babel-plugin-transform-polyfills/polyfills/String/prototype/startsWith';

const text = 'this is a string';
if (_String_prototype_startsWith(text, 'a') === true) {
  console.log('success');
} else {
  console.log('failure');
}"
`;

exports[`String.prototype.startsWith should work when part of a call chain 1`] = `
"import _String_prototype_startsWith from 'babel-plugin-transform-polyfills/polyfills/String/prototype/startsWith';

const text = 'this is a string';
if (_String_prototype_startsWith(text.substr(5), 'a')) {
  console.log('success');
} else {
  console.log('failure');
}"
`;
